### ISSUE TITLE
Issue with npm install

### ISSUE THREAD

**Issue by Kent27** [Sharcoux/slider#96]
Hi I'm trying to npm install and got this error

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: rn9@0.0.1
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"18.2.0" from the root project
npm ERR!   peer react@">=16.13.1" from @react-native-assets/slider@7.1.2
npm ERR!   node_modules/@react-native-assets/slider
npm ERR!     @react-native-assets/slider@"7.1.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^18.3.1" from react-dom@18.3.1
npm ERR! node_modules/react-dom
npm ERR!   peer react-dom@">=16.13.1" from @react-native-assets/slider@7.1.2
npm ERR!   node_modules/@react-native-assets/slider
npm ERR!     @react-native-assets/slider@"7.1.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/kent/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kent/.npm/_logs/2024-04-29T09_29_12_485Z-debug-0.log

 I'm guessing it's related to #74 
Thanks

**Comment by Sharcoux** [comment]
The log says that you installed react 18.2.0 along with react-dom: 18.3.1

Those 2 versions are not compatible because react-dom 18.3.1 requires react 18.3.1.

Just fix the dependencies oof your project.

**Comment by Nicolis1** [comment]
@Sharcoux, dependencies aren't my strong suit, so I could be wrong here, but I don't think it's as simple as that.
Updating to react@18.3.1 breaks the dependency with react-native@latest
Launching with npx create-expo-app as of today gives react@18.2.0 and react-native@0.73.6 
I assume the react-dom dependency is coming from this package, because I don't see one in my app already.

It's unclear to me how to resolve these deps.

**Comment by Nicolis1** [comment]
Update: deps resolved. @Kent27 
The problem was in my package.json I had 
```
    "react": "^18.2.0",
 ```
 with the  `^` which was allowing version 18.3.1 to get installed, breaking the dep with react-native.
 then you'll have to do the same thing with react-dom, otherwise when installing the slider it'll default to react-dom@18.3.1.
 at the end my package.json looks like
 ```
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "^0.73.6"
```
Ultimately, I think this has very little to do with the slider package, it's just that react-native right now requires ONLY 18.2.0 and is incompatible with ^18.2.0

**Comment by Nicolis1** [comment]
@Kent27 
I think I have a fix.
Replacing `react:^18.2.0` with `react:18.2.0` and also installing react-dom@18.2.0 allows you to install this package.
react-native@latest is only compatible with version 18.2.0 of react.

If you include the `^` react-dom@18.3.1 will be installed with this package, which is incompatible with react@18.2.0
You can't just upgrade react to 18.3.1 because that would break the react-native dependency.

my package.json looks like this:

```
 "dependencies": {
  "@react-native-assets/slider": "^7.2.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.73.6",
...
}
    ```

**Comment by Sharcoux** [comment]
I'm happy you solved your problem, but this is totally unrelated to this package.

**Comment by Jobjeuh** [comment]
Correct me if I'm wrong, but a React Native package doesn't (or shouldn't) have to use react-dom. So the peerDependency in package.json for react-dom shouldn't have to be there?

**Comment by Sharcoux** [comment]
It is if you need react-native-web which is needed in the development part as part of the scripts. But you're right, we should use optionalDependency. I'll make the fix.

**Comment by Sharcoux** [comment]
Fixed in 8.0.1. Thanks for the feedback.
